src/noisepy/seis/zarrstore.py

```diff
@@ -0,0 +1,154 @@
+
+    def _parse_timespans(self, filename: str) -> DateTimeRange:
+        parts = os.path.splitext(os.path.basename(filename))[0].split("T")
+        dates = [obspy.UTCDateTime(p).datetime.replace(tzinfo=datetime.timezone.utc) for p in parts]
+        return DateTimeRange(dates[0], dates[1])
+
+    def _create_station(self, arr, name: str) -> Optional[Station]:
+        # What should we do if there's not StationXML?
+        try:
+            inventory = arr[name]["StationXML"]
+            sta, net, lon, lat, elv, loc = noise_module.sta_info_from_inv(inventory)
+            return Station(net, sta, lat, lon, elv, loc)
+        except Exception as e:
+            logger.warning(f"Missing StationXML for station {name}. {e}")
+            return None
+
+
+class ZarrCCStore(CrossCorrelationDataStore):
+    def __init__(self, directory: str) -> None:
+        super().__init__()
+        self.directory = directory
+        Path(directory).mkdir(exist_ok=True)
+
+    # CrossCorrelationDataStore implementation
+    def contains(
+        self, timespan: DateTimeRange, src_chan: Channel, rec_chan: Channel, parameters: ConfigParameters
+    ) -> bool:
+        station_pair = self._get_station_pair(src_chan, rec_chan)
+        channel_pair = self._get_channel_pair(src_chan, rec_chan)
+        logger.debug(f"station pair {station_pair} channel pair {channel_pair}")
+        contains = self._contains(timespan, station_pair, channel_pair)
+        if contains:
+            logger.info(f"Cross-correlation {station_pair} and {channel_pair} already exists")
+        return contains
+
+    def save_parameters(self, parameters: ConfigParameters):
+        fc_metadata = os.path.join(self.directory, "fft_cc_data.json")
+
+        fout = open(fc_metadata, "w")
+        # WIP actually serialize this
+        fout.write(str(parameters.__dict__))
+        fout.close()
+
+    def append(
+        self,
+        timespan: DateTimeRange,
+        src_chan: Channel,
+        rec_chan: Channel,
+        params: ConfigParameters,
+        cc_params: Dict[str, Any],
+        corr: np.ndarray,
+    ):
+        # source-receiver pair: e.g. CI.ARV_CI.BAK
+        station_pair = self._get_station_pair(src_chan, rec_chan)
+        # channels, e.g. bhn_bhn
+        channels = f"{src_chan.type.name}_{rec_chan.type.name}"
+        data = np.zeros(corr.shape, dtype=corr.dtype)
+        data[:] = corr[:]
+        self._add_aux_data(timespan, cc_params, station_pair, channels, data)
+
+    def mark_done(self, timespan: DateTimeRange):
+        self._add_aux_data(timespan, {}, PROGRESS_DATATYPE, DONE_PATH, np.zeros(0))
+
+    def is_done(self, timespan: DateTimeRange):
+        done = self._contains(timespan, PROGRESS_DATATYPE, DONE_PATH)
+        if done:
+            logger.info(f"Timespan {timespan} already computed")
+        return done
+
+    def read(self, chan1: Channel, chan2: Channel, start: datetime, end: datetime) -> np.ndarray:
```

Comment:
take a look at the latest definition in `stores.py`, there's also an implementation in `asdfstore.py`